backend/asm_targets/asm_directives_intf.ml

```diff
@@ -0,0 +1,240 @@
+      been called before on the particular section, a label declaration will be
+      emitted after declaring the section. Such labels may seem strange, but
+      they are necessary so that references (e.g. DW_FORM_ref_addr /
+      DW_FORM_sec_offset when emitting DWARF) to places that are currently at
+      the start of these sections get relocated correctly when those places
+      become not at the start (e.g. during linking). *)
+  val switch_to_section : Asm_section.t -> unit
+
+  (** Called at the beginning of the assembly generation and only if the dwarf
+      flag has been set. *)
+  val initialize : unit -> unit
+
+  (** Emit an 8-bit signed integer. There is no padding or sign extension. If
+      the [comment] is specified it will be put on the same line as the
+      integer. *)
+  val int8 : ?comment:string -> Numbers.Int8.t -> unit
+
+  (** Emit a 16-bit signed integer. There is no padding or sign extension. *)
+  val int16 : ?comment:string -> Numbers.Int16.t -> unit
+
+  (** Emit a 32-bit signed integer. There is no padding or sign extension. *)
+  val int32 : ?comment:string -> Int32.t -> unit
+
+  (** Emit a 64-bit signed integer. *)
+  val int64 : ?comment:string -> Int64.t -> unit
+
+  (** Emit an 8-bit unsigned integer. There is no padding. *)
+  val uint8 : ?comment:string -> Numbers.Uint8.t -> unit
+
+  (** Emit a 16-bit unsigned integer. There is no padding. *)
+  val uint16 : ?comment:string -> Numbers.Uint16.t -> unit
+
+  (** Emit a 32-bit unsigned integer. There is no padding. *)
+  val uint32 : ?comment:string -> Numbers.Uint32.t -> unit
+
+  (** Emit a 64-bit unsigned integer. There is no padding. *)
+  val uint64 : ?comment:string -> Numbers.Uint64.t -> unit
+
+  (* CR-soon mshinwell: Target addresses should not be signed *)
+
+  (** Emit a signed integer whose width is that of an address on the target
+      machine. There is no padding or sign extension. *)
+  val targetint : ?comment:string -> Targetint.t -> unit
+
+  (** Emit a 64-bit integer in unsigned LEB128 variable-length encoding (cf.
+      DWARF debugging information standard). *)
+  val uleb128 : ?comment:string -> Numbers.Uint64.t -> unit
+
+  (** Emit a 64-bit integer in signed LEB128 variable-length encoding. *)
+  val sleb128 : ?comment:string -> Int64.t -> unit
+
+  (** Emit a string (directly into the current section). This function does not
+      write a terminating null. *)
+  val string : ?comment:string -> string -> unit
+
+  (** Cache a string for later emission. The returned label may be used to
+      obtain the address of the string in the section. This function does not
+      emit anything. (See [emit_cached_strings], below.) If a string is supplied
+      to this function that is already in the cache (associated with the same
+      section and comment) then the previously-assigned label is returned, not a
+      new one. *)
+  val cache_string : ?comment:string -> Asm_section.t -> string -> Asm_label.t
```

Comment:
(Not sure whether it is obvious, or should
appear in the docsting: the function can
leave us in a different section.)